src/lib/cozy-client/slices/sharings.js

```diff
@@ -393,14 +391,14 @@ const getSharingForRecipient = (state, document, recipient) => {
 const getDocumentActiveSharings = (state, doctype, id) => {
   const perms = getDoctypePermissions(state, doctype)
   return [
-    ...perms.byMe.filter(
-      perm => perm.attributes.permissions.files.values.indexOf(id) !== -1
+    ...perms.byMe.filter(perm =>
+      perm.attributes.permissions.files.values.includes(id)
     ),
-    ...perms.withMe.filter(
-      perm => perm.attributes.permissions.rule0.values.indexOf(id) !== -1
+    ...perms.withMe.filter(perm =>
+      perm.attributes.permissions.rule0.values.includes(id)
     )
   ]
-    .map(p => getSharing(state, p.attributes.source_id))
+    .map(p => getSharing(state, p.attributes.source_id.split('/')[1]))
```

Comment:
This is not great at all. @goldoraf are you working on this or should I work out something better here?

Reply:
Those permissions and sharing properties are very hard to follow. See https://cozy.github.io/cozy-stack/sharing.html

If I look at the `p.attributes.source_id.split('/')[1]`, I'll find `ce8835a061d0ef68947afe69a0046722` which exists three times on the response's json:

```json
{
  "data": {
    "type": "io.cozy.sharings",
    "id": "ce8835a061d0ef68947afe69a0046722",
    "meta": {
      "rev": "1-4859c6c755143adf0838d225c5e97882"
    },
    "attributes": {
      "sharing_type": "one-shot",
      "description": "sharing test",
      "preview_path": "/sharings/preview",
      "app_slug": "cal",
      "owner": true,
      "created_at": "2018-01-04T12:35:08Z",
      "updated_at": "2018-01-04T13:45:43Z"
    },
    "links": {
      "self": "/sharings/ce8835a061d0ef68947afe69a0046722"
    },
    "relationships": {
      "permissions": {
        "data": {
          "id": "46b25ad6-d044-11e7-af96-579d7e6c689e",
          "type": "io.cozy.permissions"
        }
      },
      "recipients": {
        "data": [
          {
            "id": "2a31ce0128b5f89e40fd90da3f014087",
            "type": "io.cozy.contacts",
            "status": "pending"
          }
        ]
      }
    }
  },
  "included": [
    {
      "type": "io.cozy.permissions",
      "id": "46b25ad6-d044-11e7-af96-579d7e6c689e",
      "meta": {
        "rev": "1-fbed00fed407"
      },
      "attributes": {
        "type": "shared-by-me",
        "source_id": "io.cozy.sharings/ce8835a061d0ef68947afe69a0046722",
        "codes": {
          "yuot7NaiaeGugh8T": "2a31ce0128b5f89e40fd90da3f014087"
        },
        "expires_at": 1483951978,
        "permissions": {
          "tests": {
            "description": "test",
            "type": "io.cozy.tests",
            "values": ["test-id"],
            "verbs": ["GET"]
          }
        }
      },
      "links": {
        "self": "/permissions/46b25ad6-d044-11e7-af96-579d7e6c689e"
      }
    },
    {
      "type": "io.cozy.contacts",
      "id": "2a31ce0128b5f89e40fd90da3f014087",
      "meta": {
        "rev": "1-461114b45855dc6acdb9bdc5d67e1092"
      },
      "attributes": {
        "email": {
          "address": "toto@fr"
        },
        "cozy": {
          "url": "url.fr"
        }
      },
      "links": {
        "self": "/contacts/2a31ce0128b5f89e40fd90da3f014087"
      }
    }
  ]
}
```